# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/RangeUtils.h
# Commit: 4f30d7afdf22
# Full Hash: 4f30d7afdf22c652481513564b60b37053c0948b
# Author: Atila Butkovits <abutkovits@mozilla.com>
# Date: 2025-05-08 04:32:48
# Regressor Bug: 1932150
# File Overlap Count: 1
# Description:
#   Revert "Bug 1932150 - Fix build bustage" for causing build bustages.
#   
#   This reverts commit 0681b181d6a8a48ac460a97dacb7fca91f9989c3.
#   
#   Revert "Bug 1932150 - Fix lint failure" for causing Build bustage.
# ==============================================================================

diff -r 1e64a48e3517 -r 4f30d7afdf22 dom/base/RangeUtils.h
--- a/dom/base/RangeUtils.h	Wed May 07 20:11:46 2025 +0300
+++ b/dom/base/RangeUtils.h	Wed May 07 20:37:37 2025 +0300
@@ -11,7 +11,6 @@
 #include "mozilla/RangeBoundary.h"
 #include "nsIContent.h"
 #include "nsINode.h"
-#include "nsContentUtils.h"
 
 namespace mozilla {
 
@@ -26,29 +25,23 @@
 struct ShadowDOMSelectionHelpers {
   ShadowDOMSelectionHelpers() = delete;
 
-  static nsINode* GetStartContainer(
-      const AbstractRange* aRange,
-      AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
+  static nsINode* GetStartContainer(const AbstractRange* aRange,
+                                    bool aAllowCrossShadowBoundary);
 
-  static uint32_t StartOffset(
-      const AbstractRange* aRange,
-      AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
+  static uint32_t StartOffset(const AbstractRange* aRange,
+                              bool aAllowCrossShadowBoundary);
 
-  static nsINode* GetEndContainer(
-      const AbstractRange* aRange,
-      AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
+  static nsINode* GetEndContainer(const AbstractRange* aRange,
+                                  bool aAllowCrossShadowBoundary);
 
-  static uint32_t EndOffset(
-      const AbstractRange* aRange,
-      AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
+  static uint32_t EndOffset(const AbstractRange* aRange,
+                            bool aAllowCrossShadowBoundary);
 
-  static nsINode* GetParentNodeInSameSelection(
-      const nsINode& aNode,
-      AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
+  static nsINode* GetParentNodeInSameSelection(nsINode& aNode,
+                                               bool aAllowCrossShadowBoundary);
 
-  static ShadowRoot* GetShadowRoot(
-      const nsINode* aNode,
-      AllowRangeCrossShadowBoundary aAllowCrossShadowBoundary);
+  static ShadowRoot* GetShadowRoot(const nsINode* aNode,
+                                   bool aAllowCrossShadowBoundary);
 };
 }  // namespace dom
 
@@ -56,6 +49,8 @@
   using AbstractRange = dom::AbstractRange;
 
  public:
+  static nsINode* GetParentNodeInSameSelection(const nsINode* aNode);
+
   /**
    * GetRawRangeBoundaryBefore() and GetRawRangeBoundaryAfter() retrieve
    * RawRangeBoundary which points before or after aNode.
@@ -136,9 +131,6 @@
   /**
    * The caller needs to ensure aNode is in the same doc like aAbstractRange.
    */
-  template <TreeKind aKind = TreeKind::ShadowIncludingDOM,
-            typename = std::enable_if_t<aKind == TreeKind::ShadowIncludingDOM ||
-                                        aKind == TreeKind::Flat>>
   static Maybe<bool> IsNodeContainedInRange(nsINode& aNode,
                                             AbstractRange* aAbstractRange);
 
@@ -147,18 +139,12 @@
    * ends after a range.  If neither it is contained inside the range.
    * Note that callers responsibility to ensure node in same doc as range.
    */
-  template <TreeKind aKind = TreeKind::ShadowIncludingDOM,
-            typename = std::enable_if_t<aKind == TreeKind::ShadowIncludingDOM ||
-                                        aKind == TreeKind::Flat>>
   static nsresult CompareNodeToRange(nsINode* aNode,
                                      AbstractRange* aAbstractRange,
                                      bool* aNodeIsBeforeRange,
                                      bool* aNodeIsAfterRange);
 
-  template <TreeKind aKind, typename SPT, typename SRT, typename EPT,
-            typename ERT,
-            typename = std::enable_if_t<aKind == TreeKind::ShadowIncludingDOM ||
-                                        aKind == TreeKind::Flat>>
+  template <typename SPT, typename SRT, typename EPT, typename ERT>
   static nsresult CompareNodeToRangeBoundaries(
       nsINode* aNode, const RangeBoundaryBase<SPT, SRT>& aStartBoundary,
       const RangeBoundaryBase<EPT, ERT>& aEndBoundary, bool* aNodeIsBeforeRange,